src/main/java/software/amazon/awssdk/crt/s3/ChecksumAlgorithm.java

```diff
@@ -45,4 +50,27 @@ private static Map<Integer, ChecksumAlgorithm> buildEnumMapping() {
     private int nativeValue;
 
     private static Map<Integer, ChecksumAlgorithm> enumMapping = buildEnumMapping();
+
+    /**
+     * @hidden Marshals a list of algorithm into an array for Jni to deal with
+     *
+     * @param algorithms list of algorithms
+     * @return a int[] that with the [algorithms.nativeValue, *]
+     */
+    public static int[] marshallAlgorithmsForJNI(final List<ChecksumAlgorithm> algorithms) {
+        if (algorithms == null) {
```

Comment:
style suggestion: this whole method can be something like
Optional.ofNullable(algorithms).map(ChecksumAlgorithm::getNativeValue).toArray().orElseDefault(null);

Reply:
did it a bit different by adding another function to convert the list to an int array. Thanks for the suggestion!